### PR TITLE
Factor out model updates and move them to mesh-models

### PR DESCRIPTION
**Pull request by kraman**
Connect to strongloop/strong-pm#131

### CODE REVIEW

**Comment by kraman** [comment]
Requires https://github.com/strongloop/strong-mesh-models/pull/20

**Comment by rmg** [comment]
The first two commits look like they form 1 logical commit. Was that to make reviewing easier?

I know it's probably a lot to ask, but is there any chance those commits could be broken up into one commit per handler move?

**Comment by kraman** [comment]
@rmg tried to break it up into more manageable chunks for you

**Comment by kraman** [comment]
@slnode test please

File: lib/server.js [hunk 1]
```diff
@@ -92,459 +92,114 @@ function Server(originalCommand, baseDir, listenPort, controlPath) {
...
-    if (!proc) {
-      debug('Ignoring metrics for unknown worker pid: %d', pid);
-      return deleteOld();
-    }
+Server.prototype._onMasterStart =
+  function _onMasterStart(supervisorInfo, callback) {
+    var self = this;
+    var current = runner.current();
+    var commit = current.commit;
+    var strongPmInfo = require(path.join(__dirname, '..', 'package.json'));
```
Comment (rmg) [inline lib/server.js]: Should be shortened to `require('../package.json')`.

File: lib/server.js [hunk 2]
```diff
@@ -92,459 +92,114 @@ function Server(originalCommand, baseDir, listenPort, controlPath) {
...
+        supervisorInfo.id = 0; // Supervisor is worker id 0
+        supervisorInfo.ppid = process.pid;
+
+        // Instance information
+        supervisorInfo.commitHash = commit ? commit.hash : undefined;
+        supervisorInfo.npmModules = appInfo;
+        supervisorInfo.PMPort = self._listenPort;
+        supervisorInfo.containerVersionInfo = {
+          os: {
+            platform: os.platform(),
```
Comment (rmg) [inline lib/server.js]: Would you mind putting a `TODO` comment in here to make sure we don't miss then when we refactor the runner into drivers? Containerized pm vs. containerized app comes to mind as a case where this code will break as is.

**Comment by rmg** [comment]
Just a few minor tweaks and I think it'll be ready to merge.

**Comment by rmg** [comment]
LGTM.

**Pull request merged**